moshi/src/test/java/com/squareup/moshi/TypesTest.java

```diff
@@ -292,6 +300,31 @@ public void arrayEqualsGenericTypeArray() {
     assertThat(Types.equals(Types.arrayOf(String.class), String[].class)).isTrue();
   }
 
+  private E<A, B> methodReturningE() {
+    throw new Error(); // Intentionally not implemented
```

Comment:
nit: use `NotImplementedException`?

Reply:
Makes sense to me, I used NotImplementedError from Kotlin, since that shouldn't be checked since it extends Error instead of Exception.